tfhe/src/core_crypto/commons/math/random/mod.rs

```diff
@@ -67,9 +67,9 @@ where
         distribution: D,
         slice: &mut [Self],
     ) {
-        slice.iter_mut().for_each(|s| {
+        for s in slice.iter_mut() {
             *s = Self::generate_one(generator, distribution);
-        });
+        }
```

Comment:
what motivated that change ?

Reply:
using for_each is an 'anti-pattern', you should always (unless its not possible) prefer for loop instead of using for_each

> This is equivalent to using a [for](https://doc.rust-lang.org/book/ch03-05-control-flow.html#looping-through-a-collection-with-for) loop on the iterator, although break and continue are not possible from a closure. *It’s generally more idiomatic to use a for loop*

https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.for_each

Reply:
huh, I thought the for_each approach was preferred for some iterator specific optimization 🤔

Reply:
https://rust-lang.github.io/rust-clippy/master/index.html#/needless_for_each

Reply:
ok yeah then my bad, I started to adopt the "for each is generally better", which seems to be the case unless you have only a single variable or need some break/continue functionality